src/sinks/pulsar.rs

```diff
@@ -249,6 +268,12 @@ fn encode_event(mut item: Event, encoding: &EncodingConfig<Encoding>) -> crate::
             .get(log_schema().message_key())
             .map(|v| v.as_bytes().to_vec())
             .unwrap_or_default(),
+        Encoding::Avro => {
+            let schema = avro_rs::Schema::parse_str(encoding.schema().as_ref().unwrap()).unwrap();
```

Comment:
I'm curious how long parsing the schema takes. Perhaps we should be storing the built schema at the sink level so it can be used repeatedly?

Reply:
Yup, this was my point at https://github.com/timberio/vector/pull/5021#issuecomment-726785043, but put it to EncodingCondig is quite a lot of work. As Schema depends on transforms applied, I agree with you to put the built schema at the sink level.

Reply:
Done 160ffcb at the sink level, not sure how to do it using log_schema.

Reply:
Nicee!!!! It looks much better. :)